src/cookie-consent.mjs

```diff
@@ -42,6 +42,14 @@ const CookieConsent = settings => {
   if (preferences.hasPreferences()) {
     events.dispatch('update', preferences.getAll());
   } else {
+    // Fire the `update` event for required cookies regardless of consent.
+    if (settings.cookies) {
+      const requiredCookies = settings.cookies.filter(cookie => cookie.required).map(cookie => ({
+        id: cookie.id,
+        accepted: true,
+      }));
+      events.dispatch('update', requiredCookies);
```

Comment:
You should actually check `requiredCookies.length` to avoid an `update` event with empty payload.